### PR TITLE
Update build process

### PR DESCRIPTION
**Pull request by dmitryk-dk**
Implement build and release process

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	> victoriametrics-datasource-$(PKG_TAG)_checksums.txt \
+	&& rm -rf ./dist
+
+
+victoriametrics-datasource-plugin-release: \
+	victorimetrics-frontend-plugin-build \
+	victoriametrics-backend-plugin-build \
+	victoriametrics-datasource-plugin-pack
+
+build-release:
```
Comment (hagen1778) [inline Makefile:78]: Please, update README with info about making a new release and use of Makefile commands.

File: README.md [hunk 2]
```diff
@@ -79,6 +79,66 @@ yarn dev
 yarn build:zip
 ```
 
+### 3. Run the backend plugin
+
+From the root folder of the project you can run
```
Comment (hagen1778) [inline README.md:84]:
```suggestion
From the root folder of the project run the following command:
```

File: README.md [hunk 3]
```diff
@@ -79,6 +79,66 @@ yarn dev
...
+make victoriametrics-backend-plugin-linux-386-prod
+```
+* amd64
+```
+make victoriametrics-backend-plugin-amd64-prod
+```
+* arm64
+```
+make victoriametrics-backend-plugin-arm64-prod
+```
```
Comment (hagen1778) [inline README.md:120]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -79,6 +79,66 @@ yarn dev
...
+
+To use backend plugin you should define in the `plugin.json` file:
+
+1. "backend": true
+2. "executable": "victoriametrics_datasource_plugin"
+
+And run your grafana server.
```
Comment (hagen1778) [inline README.md:127]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -79,6 +79,66 @@ yarn dev
 yarn build:zip
 ```
 
+### 3. Run the backend plugin
```
Comment (hagen1778) [inline README.md:82]:
```suggestion
### 3. How to build backend plugin
```

File: README.md [hunk 6]
```diff
@@ -79,6 +79,66 @@ yarn dev
...
+```
+
+To use backend plugin you should define in the `plugin.json` file:
+
+1. "backend": true
+2. "executable": "victoriametrics_datasource_plugin"
+
+And run your grafana server.
+
+## Release version
```
Comment (hagen1778) [inline README.md:129]:
```suggestion
## How to make new release
```

File: README.md [hunk 7]
```diff
@@ -79,6 +79,66 @@ yarn dev
...
+
+0. Make sure that the release commits have no security issues.
```
Comment (hagen1778) [inline README.md:131]:
```suggestion
0. Make sure there are no open security issues.
```

File: README.md [hunk 8]
```diff
@@ -79,6 +79,66 @@ yarn dev
...
+1. Create the following release tags:
```
Comment (hagen1778) [inline README.md:132]:
```suggestion
1. Create a release tag:
```

File: README.md [hunk 9]
```diff
@@ -79,6 +79,66 @@ yarn dev
...
+    * `git tag -s v1.xx.y` in `master` branch
+2. Run `TAG=v1.xx.y make build-release`. This command performs the following tasks:
+   a) Build and package binaries in `*.tar.gz` release archives with the corresponding `_checksums.txt` files inside `dist` directory.
+   This step can be run manually with the command `make build-release` from the needed git tag.
```
Comment (hagen1778) [inline README.md:136]:
```suggestion
2. Run `TAG=v1.xx.y make build-release` to build and package binaries in `*.tar.gz` release archives.
```

File: README.md [hunk 10]
```diff
@@ -79,6 +79,66 @@ yarn dev
...
+3. Push the tags created `v1.xx.y` at step 2 to public GitHub repository
```
Comment (hagen1778) [inline README.md:137]: Please provide a command example for this step

File: README.md [hunk 11]
```diff
@@ -79,6 +79,66 @@ yarn dev
...
+4. Go to <https://github.com/VictoriaMetrics/grafana-datasource/releases> and verify that draft release with the name `TAG` has been created
+   and this release contains all the needed binaries and checksums.
+5. Remove the `draft` checkbox for the `TAG` release and manually publish it.
```
Comment (hagen1778) [inline README.md:140]: I believe, this step misses creation of https://github.com/VictoriaMetrics/grafana-datasource/blob/main/victoriametrics-datasource-v0.1.1.zip and its checksum

Comment (dmitryk-dk) [reply README.md:140]: The backend and frontend build in one folder, `dist`. In that case, I prepared the build process which can build both parts into one tar.gz archive. I we should split this operation I can update make file

Comment (dmitryk-dk) [reply README.md:121]: Added `frontend-build-release` and `victoriametrics-datasource-frontend-plugin-release` commands which build only the frontend and pack it into `zip` format.

**Review by hagen1778 — APPROVED** [review]

**Comment by hagen1778** [comment]
Thank you!

**Pull request merged**